src/main/java/com/kolown/porring/board/service/ReactionService.java

```diff
@@ -4,38 +4,68 @@
 import com.kolown.porring.board.dto.request.ReactionRequestDto;
 import com.kolown.porring.board.entity.Board;
 import com.kolown.porring.board.entity.Reaction;
+import com.kolown.porring.board.entity.ReactionType;
 import com.kolown.porring.board.repository.BoardRepository;
 import com.kolown.porring.board.repository.ReactionRepository;
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
+import java.util.HashSet;
 import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import org.springframework.web.client.HttpClientErrorException;
 
 @Service
 @RequiredArgsConstructor
 public class ReactionService {
     private final ReactionRepository reactionRepository;
     private final BoardRepository boardRepository;
 
-    public List<Reaction> getAllReactionsByBoardId(Long boardId) {
-        return reactionRepository.findByIdBoardId(boardId);
-    }
+    public List<ReactionType> getBoardReactionList(Long boardId) {
 
-    public Reaction createReaction(Account account, long boardId, ReactionRequestDto reactionRequestDto) {
-        Board board = boardRepository.findById(boardId).orElseThrow();
+        Set<ReactionType> reactionMap = new HashSet<>();
 
-        Reaction reaction = new Reaction(board, account, reactionRequestDto.getReactionType());
+        reactionRepository.findByBoardId(boardId).forEach(reaction -> {
+            reactionMap.add(reaction.getReactionType());
+        });
+
+        return new ArrayList<>(reactionMap);
+    }
 
-        return reactionRepository.save(reaction);
+    public ReactionType getBoardReactionOfAccount(Long boardId, Account account) {
+        return reactionRepository
+            .findByAccountIdAndBoardId(account.getId(), boardId)
+            .orElse(null)
+            .getReactionType();
     }
 
-    public Reaction updateBoardReaction(Long boardId, Long accountId, ReactionRequestDto reactionRequestDto) {
-        Reaction.ReactionId id = new Reaction.ReactionId(boardId, accountId);
+    public void updateBoardReaction(
+        Long boardId,
+        Account account,
+        ReactionRequestDto reactionRequestDto
+    ) {
+        Board board = boardRepository.findById(boardId).orElseThrow();
+
+        Reaction.ReactionId id = new Reaction.ReactionId(boardId, account.getId());
+
+        if (reactionRequestDto.getReactionType() == null) {
+            reactionRepository.deleteById(id);
+            return;
+        }
 
-        Reaction reaction = reactionRepository.findById(id).orElseThrow();
+        Optional<Reaction> reaction = reactionRepository
+            .findByAccountIdAndBoardIdWithDeleted(boardId, account.getId());
 
-        reaction.setReactionType(reactionRequestDto.getReactionType());
+        if (!reaction.isPresent()) {
```

Comment:
```suggestion
        if (reaction.isEmpty()) {
```

p5: '옵셔널의 값이 비었는가?'를 확인할 수 있는 명확한 코드로 변경해보면 좋을 것 같아요. `isEmpty`였는지는 잘 기억나지 않네요.